### ISSUE TITLE
Issues with Psych v4.0.0

### ISSUE THREAD

**Issue by jrgarcia** [sds/overcommit#752]
I started having an issue with `overcommit` v0.57.0 recently and upon looking into it, the issue appears to be with the latest version of `psych` v4.0.0. Here is a stack trace:

```
Traceback (most recent call last):
	53: from /Users/my-user/.rbenv/versions/2.7.2/bin/overcommit:23:in `<main>'
	52: from /Users/my-user/.rbenv/versions/2.7.2/bin/overcommit:23:in `load'
	51: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/overcommit-0.57.0/bin/overcommit:50:in `<top (required)>'
	50: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/overcommit-0.57.0/lib/overcommit/cli.rb:20:in `run'
	49: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/overcommit-0.57.0/lib/overcommit/cli.rb:48:in `parse_arguments'
	48: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/2.7.0/optparse.rb:1691:in `parse!'
	47: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/2.7.0/optparse.rb:1666:in `permute!'
	46: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/2.7.0/optparse.rb:1569:in `order!'
	45: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/2.7.0/optparse.rb:1575:in `parse_in_order'
	44: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/2.7.0/optparse.rb:1575:in `catch'
	43: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/2.7.0/optparse.rb:1621:in `block in parse_in_order'
	42: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/overcommit-0.57.0/lib/overcommit/cli.rb:80:in `block in add_information_options'
	41: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/overcommit-0.57.0/lib/overcommit/cli.rb:157:in `print_installed_hooks'
	40: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/overcommit-0.57.0/lib/overcommit/cli.rb:220:in `config'
	39: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/overcommit-0.57.0/lib/overcommit/configuration_loader.rb:57:in `load_repo_config'
	38: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/overcommit-0.57.0/lib/overcommit/configuration_loader.rb:66:in `load_file'
	37: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/overcommit-0.57.0/lib/overcommit/configuration_loader.rb:15:in `default_configuration'
	36: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/overcommit-0.57.0/lib/overcommit/configuration_loader.rb:28:in `load_from_file'
	35: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych.rb:586:in `safe_load_file'
	34: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych.rb:586:in `open'
	33: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych.rb:587:in `block in safe_load_file'
	32: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych.rb:334:in `safe_load'
	31: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:35:in `accept'
	30: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/visitor.rb:6:in `accept'
	29: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/visitor.rb:30:in `visit'
	28: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:318:in `visit_Psych_Nodes_Document'
	27: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:35:in `accept'
	26: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/visitor.rb:6:in `accept'
	25: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/visitor.rb:30:in `visit'
	24: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:167:in `visit_Psych_Nodes_Mapping'
	23: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:343:in `revive_hash'
	22: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:343:in `each_slice'
	21: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:343:in `each'
	20: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:345:in `block in revive_hash'
	19: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:35:in `accept'
	18: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/visitor.rb:6:in `accept'
	17: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/visitor.rb:30:in `visit'
	16: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:167:in `visit_Psych_Nodes_Mapping'
	15: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:343:in `revive_hash'
	14: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:343:in `each_slice'
	13: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:343:in `each'
	12: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:345:in `block in revive_hash'
	11: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:35:in `accept'
	10: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/visitor.rb:6:in `accept'
	 9: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/visitor.rb:30:in `visit'
	 8: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:167:in `visit_Psych_Nodes_Mapping'
	 7: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:343:in `revive_hash'
	 6: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:343:in `each_slice'
	 5: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:343:in `each'
	 4: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:345:in `block in revive_hash'
	 3: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:35:in `accept'
	 2: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/visitor.rb:6:in `accept'
	 1: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/visitor.rb:30:in `visit'
/Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:430:in `visit_Psych_Nodes_Alias': Unknown alias: chamber_settings_files (Psych::BadAlias)
	53: from /Users/my-user/.rbenv/versions/2.7.2/bin/overcommit:23:in `<main>'
	52: from /Users/my-user/.rbenv/versions/2.7.2/bin/overcommit:23:in `load'
	51: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/overcommit-0.57.0/bin/overcommit:50:in `<top (required)>'
	50: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/overcommit-0.57.0/lib/overcommit/cli.rb:20:in `run'
	49: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/overcommit-0.57.0/lib/overcommit/cli.rb:48:in `parse_arguments'
	48: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/2.7.0/optparse.rb:1691:in `parse!'
	47: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/2.7.0/optparse.rb:1666:in `permute!'
	46: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/2.7.0/optparse.rb:1569:in `order!'
	45: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/2.7.0/optparse.rb:1575:in `parse_in_order'
	44: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/2.7.0/optparse.rb:1575:in `catch'
	43: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/2.7.0/optparse.rb:1621:in `block in parse_in_order'
	42: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/overcommit-0.57.0/lib/overcommit/cli.rb:80:in `block in add_information_options'
	41: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/overcommit-0.57.0/lib/overcommit/cli.rb:157:in `print_installed_hooks'
	40: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/overcommit-0.57.0/lib/overcommit/cli.rb:220:in `config'
	39: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/overcommit-0.57.0/lib/overcommit/configuration_loader.rb:57:in `load_repo_config'
	38: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/overcommit-0.57.0/lib/overcommit/configuration_loader.rb:66:in `load_file'
	37: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/overcommit-0.57.0/lib/overcommit/configuration_loader.rb:15:in `default_configuration'
	36: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/overcommit-0.57.0/lib/overcommit/configuration_loader.rb:28:in `load_from_file'
	35: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych.rb:586:in `safe_load_file'
	34: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych.rb:586:in `open'
	33: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych.rb:587:in `block in safe_load_file'
	32: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych.rb:334:in `safe_load'
	31: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:35:in `accept'
	30: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/visitor.rb:6:in `accept'
	29: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/visitor.rb:30:in `visit'
	28: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:318:in `visit_Psych_Nodes_Document'
	27: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:35:in `accept'
	26: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/visitor.rb:6:in `accept'
	25: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/visitor.rb:30:in `visit'
	24: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:167:in `visit_Psych_Nodes_Mapping'
	23: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:343:in `revive_hash'
	22: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:343:in `each_slice'
	21: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:343:in `each'
	20: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:345:in `block in revive_hash'
	19: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:35:in `accept'
	18: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/visitor.rb:6:in `accept'
	17: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/visitor.rb:30:in `visit'
	16: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:167:in `visit_Psych_Nodes_Mapping'
	15: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:343:in `revive_hash'
	14: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:343:in `each_slice'
	13: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:343:in `each'
	12: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:345:in `block in revive_hash'
	11: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:35:in `accept'
	10: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/visitor.rb:6:in `accept'
	 9: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/visitor.rb:30:in `visit'
	 8: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:167:in `visit_Psych_Nodes_Mapping'
	 7: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:343:in `revive_hash'
	 6: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:343:in `each_slice'
	 5: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:343:in `each'
	 4: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:345:in `block in revive_hash'
	 3: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:35:in `accept'
	 2: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/visitor.rb:6:in `accept'
	 1: from /Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/visitor.rb:30:in `visit'
/Users/my-user/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/psych-4.0.0/lib/psych/visitors/to_ruby.rb:430:in `visit_Psych_Nodes_Alias': Unable to load configuration from '/Users/my-user/code/my-repo/.overcommit.yml': Unknown alias: chamber_settings_files (Overcommit::Exceptions::ConfigurationError)
```

Downgrading `psych` to the previous version (v3.3.2) got it working again.

**Comment by sds** [comment]
This is a breaking behavior change with the upstream Psych gem: https://bugs.ruby-lang.org/issues/17866

Looks like other projects (e.g. [Rails](https://github.com/rails/rails/pull/42249)) are just working around it. We'll do the same.

Fixed in 10c96ebe5c1bf640c53cbac4362df1dd943d4b84. Thanks for the report!

**Comment by jrgarcia** [comment]
Thanks so much for the quick response!
